acquia_lift.admin.inc

```diff
@@ -1241,3 +1242,58 @@ function acquia_lift_get_option_set_for_targeting($agent_name) {
   }
   return reset($option_sets);
 }
+
+/**
+ * Validate that an agent has the minimum targeting required and add it if it
+ * is not available.
+ *
+ * Targeting agents should have a default "Everyone else" audience that has all
+ * unassigned variations assigned.  Variations should only be automatically
+ * added when the agent has not been started.
+ *
+ * @param stdClass $agent
+ *   The object representing the agent that owns the targeting option set.
+ */
+function acquia_lift_validate_minimum_targeting($agent) {
+  if (personalize_agent_get_status($agent->machine_name) !== PERSONALIZE_STATUS_NOT_STARTED) {
+    return;
+  }
+  $option_set = acquia_lift_get_option_set_for_targeting($agent->machine_name);
+  if (empty($option_set->targeting)) {
+    $audience = _acquia_lift_personalize_campaign_wizard_everyone_else_audience();
+    acquia_lift_target_audience_save($audience['name'], $agent->machine_name, $audience['contexts'], $audience['strategy'], $audience['weight'], $audience['id']);
+    $assigned = array();
+  }
+  else {
+    // Find all of the options that are assigned to other audiences.
+    $existing = acquia_lift_get_structure_from_targeting($option_set);
+    $assigned = array();
+    foreach ($existing as $audience_id => $options) {
+      if ($audience_id === ACQUIA_LIFT_TARGETING_EVERYONE_ELSE) {
+        continue;
+      }
+      $assigned = array_merge($assigned, $options);
+    }
+  }
+  $targeting_options = array();
+  foreach ($option_set->options as $option) {
+    $targeting_options[] = $option['option_id'];
+  }
+  // Any unassigned option is automatically assigned to the "Everyone else"
+  // category.
+  $unassigned_options = array_diff($targeting_options, $assigned);
+  acquia_lift_save_targeting_structure($agent, array(ACQUIA_LIFT_TARGETING_EVERYONE_ELSE => $unassigned_options));
```

Comment:
You're right - this is backwards.  It may only really need to check the agent property rather than the option set property because by the time that is populated, you are starting the agent and shouldn't be automatically changing things anyway.